### ISSUE TITLE
CommonHTML problems with invalid MathML [was: Bug: MathJax v2.5 beta Fast Preview feature not rendering boxed expressions]

### ISSUE THREAD

**Issue by ddws** [mathjax/MathJax#1006]
In testing the new preview output feature (Fast Preview) in MathJax v2.5 beta, we found that boxed expressions within a table cell do not display. If the Fast Preview feature is turned off, MathJax v2.5 beta is able to render boxed expressions: 

![capture](https://cloud.githubusercontent.com/assets/10518108/5748380/76130a50-9c0d-11e4-8d9b-9504a2a17a47.JPG)

But when the Fast Preview feature is turned back on, Math Processing Error messages appear:

![capture1](https://cloud.githubusercontent.com/assets/10518108/5748924/e9740de2-9c11-11e4-9cd3-a3841d6e4bcf.JPG)

Is it possible to fix the Fast Preview feature in MathJax v2.5 beta to render and display a boxed expression? The browser I used was Internet Explorer 8.

Below is the HTML code for the boxed expression in the screen shots above:

<math id="m256" display="inline" xmlns="http://www.w3.org/1998/Math/MathML">
<mrow>
  <mtable frame="solid">
    <mtd>
      <mrow>
        <mn>0.10677</mn>
        <mi>&#x2009;&#x200A;&#x200A;</mi>
        <mi>&#x2009;&#x200A;&#x200A;</mi>
        <mi>&#x2009;&#x200A;&#x200A;</mi>
        <mn>231</mn>
      </mrow>
    </mtd>
  </mtable>
</mrow>
</math>

**Comment by ddws** [comment]
<math id="m256" display="inline" xmlns="http://www.w3.org/1998/Math/MathML">
<mrow>
  <mtable frame="solid">
    <mtd>
      <mrow>
        <mn>0.10677</mn>
        <mi>&#x2009;&#x200A;&#x200A;</mi>
        <mi>&#x2009;&#x200A;&#x200A;</mi>
        <mi>&#x2009;&#x200A;&#x200A;</mi>
        <mn>231</mn>
      </mrow>
    </mtd>
  </mtable>
</mrow>
</math>

**Comment by pkra** [comment]
Thanks for the report. Did you accidentally close the issue? I can reproduce it in any case.

**Comment by pkra** [comment]
The problem seems to originate in the missing `<mtr>`. So

```html
<math id="m256" display="inline" xmlns="http://www.w3.org/1998/Math/MathML">
<mrow>
  <mtable frame="solid">
<mtr>
    <mtd>
      <mrow>
        <mn>0.10677</mn>
        <mi>&#x2009;&#x200A;&#x200A;</mi>
        <mi>&#x2009;&#x200A;&#x200A;</mi>
        <mi>&#x2009;&#x200A;&#x200A;</mi>
        <mn>231</mn>
      </mrow>
    </mtd>
<\mtr>
  </mtable>
</mrow>
</math>
```

should not produce a rendering issue (though the CommonHTML output does not yet create the border).

Strangely enough, `toMathML` gives me

```html
<math xmlns="http://www.w3.org/1998/Math/MathML" display="inline" id="m256">
  <mrow>
    <mtable frame="solid">
      <mtd>
        <mtd>
          <mrow>
            <mn>0.10677</mn>
            <mi>&#x2009;&#x200A;&#x200A;</mi>
            <mi>&#x2009;&#x200A;&#x200A;</mi>
            <mi>&#x2009;&#x200A;&#x200A;</mi>
            <mn>231</mn>
          </mrow>
        </mtd>
      </mtd>
    </mtable>
  </mrow>
</math>
```

which seems worse.

Even more strangely,  [the W3C validator](http://validator.w3.org/check) validates this as MathML2 -- though at least the HTML5 validator will correctly complain that [the spec says](http://www.w3.org/Math/draft-spec/chapter3.html#presm.mtable) the children of `<mtable>` must be `<mtr>`s.

Anyway, something fishy is going on.

**Comment by dpvc** [comment]
Since the children of `<mtable>` must be `<mtr>` or `<mlabeledtr>`, MathJax attempts to force that by inserting `<mtr>` if there isn't one.  But there was an error where it was inserting `<mtd>` rather than `<mtr>` (probably I copied and pasted from the corresponding code that forces children of `<mtr>` to be `<mtd>`).

The `issue1006` branch of my fork of MathJax resolves the issue.

**Comment by dpvc** [comment]
==> Merged (into hotfix-2.5)

**Comment by dpvc** [comment]
=> In testsuite

Parsing/issue1006.html
